tests/unit/components/Items/Question/Body.spec.js

```diff
@@ -2,266 +2,348 @@ import { mount } from "@vue/test-utils";
-      props: {
-        questionText: testQuestionText,
-      },
+  describe("generic", () => {
+    it("should render with default values", () => {
+      expect(wrapper.find('[data-test="optionContainer"]').exists()).toBe(true);
+      expect(
+        wrapper.find('[data-test="subjectiveAnswerContainer"]').exists()
+      ).toBe(false);
     });
-    expect(wrapper.find('[data-test="questionText"]').text()).toBe(
-      testQuestionText
-    );
-  });
 
-  it("renders options", () => {
-    const options = ["a", ""];
-    const wrapper = mount(Body, {
-      props: {
-        options: options,
-      },
+    it("renders question text", async () => {
+      const testQuestionText = "ABC";
+      await wrapper.setProps({
+        questionText: testQuestionText,
+      });
+      expect(wrapper.find('[data-test="questionText"]').text()).toBe(
+        testQuestionText
+      );
     });
-    expect(wrapper.find('[data-test="option-0"]').exists()).toBe(true);
-    expect(wrapper.find('[data-test="option-1"]').exists()).toBe(true);
-    expect(wrapper.find('[data-test="option-2"]').exists()).toBe(false);
-    expect(wrapper.find('[data-test="option-0"]').text()).toBe(options[0]);
-  });
 
-  it("option text selected correctly", () => {
-    const options = ["a", ""];
-    const wrapper = mount(Body, {
-      props: {
+    it("renders mcq options", async () => {
+      const options = ["a", ""];
+      await wrapper.setProps({
         options: options,
-      },
+      });
+      expect(wrapper.find('[data-test="option-0"]').exists()).toBe(true);
+      expect(wrapper.find('[data-test="option-1"]').exists()).toBe(true);
+      expect(wrapper.find('[data-test="option-2"]').exists()).toBe(false);
+      expect(wrapper.find('[data-test="option-0"]').text()).toBe(options[0]);
+      expect(wrapper.vm.optionInputType).toBe("radio");
     });
-    wrapper.find('[data-test="option-0"]').trigger("click");
-    expect(wrapper.emitted()).toHaveProperty("option-selected");
-  });
 
-  it("option radio selected correctly", () => {
-    const options = ["a", ""];
-    const wrapper = mount(Body, {
-      props: {
-        options: options,
-      },
+    it("starts loading image if imageData is passed", () => {
+      const startImageLoading = jest.spyOn(Body.methods, "startImageLoading");
+      mount(Body, {
+        props: {
+          questionType: "mcq",
+          imageData: {
+            url: "mock",
+            alt_text: "mock",
+          },
+        },
+      });
+
+      expect(startImageLoading).toHaveBeenCalled();
     });
-    wrapper.find('[data-test="radio-0"]').trigger("click");
-    expect(wrapper.emitted()).toHaveProperty("option-selected");
   });
 
-  it("should render subjective question with default values", () => {
-    const wrapper = mount(Body, {
-      props: {
-        questionType: "subjective",
-      },
+  describe("subjective questions", () => {
+    const mountWrapper = () => {
+      wrapper = mount(Body, {
```

Comment:
So basically you're just mounting a new instance of the component and reassinging it to the wrapper variable. But doesn't this mean that the previously mounted instance will stay mounted?
If that is the case, then it will make tests unnecessarily slow. Best to unmount if we're remounting?